### PR TITLE
chore: bump vitest related deps

### PR DESCRIPTION
**Pull request by gweesin**
# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-vue/blob/dev/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

image

Issue Number: N/A

## What is the new behavior?

Performance improvement

caba0dcd2a46c22ad8d479ae997a2a7a

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by kagol** [comment]
@gweesin 建议只修改以下3个命令涉及到的 vitest 版本号：
```
    "test:unit2": "pnpm -C examples/vue2 test:unit",
    "test:unit2.7": "pnpm -C examples/vue2.7 test:unit",
    "test:unit3": "pnpm -C examples/vue3 test:unit",
```
其他无关的无需修改

File: internals/cli/package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "rollup": "^3.7.3",
     "semver": "^7.3.8",
     "tsup": "7.2.0",
-    "vite": "^4.3.8",
+    "vite": "^5.3.5",
```
Comment (zzcr) [inline internals/cli/package.json]: 你好，vite的升级，需要验证下是否对构建发布造成影响？不知道是否有验证这方面的问题

Comment (gweesin) [reply internals/cli/package.json]: 跑测试用例的结果是没变，原本报错还是报错，原本成功还是成功。publish命令我这里跑不了就没验证了

**Comment by gweesin** [comment]
> @gweesin 建议只修改以下3个命令涉及到的 vitest 版本号：
> 
> ```
>     "test:unit2": "pnpm -C examples/vue2 test:unit",
>     "test:unit2.7": "pnpm -C examples/vue2.7 test:unit",
>     "test:unit3": "pnpm -C examples/vue3 test:unit",
> ```
> 
> 其他无关的无需修改

修改这些点主要基于以下两个原则：
1. vitest 关联的运行时内核 jsdom、关联的 vite（vitest 2.x 需要 vite 5.x）统一升级
2. 统一monorepo下依赖包版本

**Comment by zzcr** [comment]
代码有冲突，可以解决下，我先合入再验证下

**Pull request closed without merging**